Update tutorials to use `create_agents` and `rng.integers`

DarshPareek opened the pull request:
### Summary
Updates the tutorials to align with Mesa 3.1's recommended practices for agent creation and random number generation.

### Motive
The changes make the tutorials consistent with Mesa 3.1 by:
- Using the `create_agents` class method instead of individual agent creation in loops
- Replacing `random.randrange` with `rng.integers` for position generation
- Improving performance by generating all agent positions at once instead of one at a time

### Implementation
Made consistent updates across `intro_tutorial.ipynb` and `MoneyModel.py`:

1. Agent Creation:
Old approach:
```python
for _ in range(self.num_agents):
    a = MoneyAgent(self)
```

New approach:
```python
agents = MoneyAgent.create_agents(model=self, n=n)
```

2. Position Generation:
Old approach:
```python
x = self.random.randrange(self.grid.width)
y = self.random.randrange(self.grid.height)
```

New approach:
```python
x = self.rng.integers(0, self.grid.width, size=(n,))
y = self.rng.integers(0, self.grid.height, size=(n,))
```

### Usage Examples
Example of the updated model initialization:
```python
class MoneyModel(mesa.Model):
    def __init__(self, n, width, height, seed=None):
        super().__init__(seed=seed)
        self.num_agents = n
        self.grid = mesa.space.MultiGrid(width, height, True)
        
        # Create agents all at once
        agents = MoneyAgent.create_agents(model=self, n=n)
        
        # Generate all positions at once
        x = self.rng.integers(0, self.grid.width, size=(n,))
        y = self.rng.integers(0, self.grid.height, size=(n,))
        
        # Place agents efficiently using zip
        for a, i, j in zip(agents, x, y):
            self.grid.place_agent(a, (i, j))
```

EwoutH commented on docs/tutorials/intro_tutorial.ipynb:
Let’s make they arguments explicit here:
```Python
create_agents(model=self, n=n)
```

quaquel replied:
I would also use self.rng to create the x and y coordinates

```python

x = self.rng.randint(0, self.grid.width, size=(n,))
y = self.rng.randint(0, self.grid.height, size=(n,))

```

EwoutH replied:
Agreed, but other PR. Let’s keep them atomic, especially with new contributors.

EwoutH replied:
@DarshPareek please don’t forget my original comment here.

EwoutH reviewed (commented):
Thanks for the PR!

EwoutH commented:
I did assign @ashish-nagmoti to issue #2537 by the way. Next time please do check if an issue is assigned, and if it is, ask first.

@ashish-nagmoti would you like to review this PR instead?

DarshPareek commented:
In the latest update I have changed the arguments to be more explicit in `create_agents` function and added 
```
x = self.rng.integers(0, self.grid.width, size=(n,)) 
y = self.rng.integers(0, self.grid.height, size=(n,))
```
to make positioning of agents better

EwoutH commented on docs/tutorials/MoneyModel.py:
```diff
@@ -62,12 +62,13 @@ def __init__(self, n=10, width=10, height=10, seed=None):
         # Create agents
-        for _ in range(self.num_agents):
-            a = MoneyAgent(self)
+        agents = MoneyAgent.create_agents(model=self,n=n)
```
It’s best practice to leave a space after each argument when calling a method. Please update this everywhere.
```suggestion
        agents = MoneyAgent.create_agents(model=self, n=n)
```

DarshPareek replied:
I will format these according to black formatter

EwoutH reviewed (approved):
I’m good, but let’s see if Jan also is.

quaquel commented:
I am fine with this, although I personally would prefer to move `self.grid.place_agent(a, (i, j))` into the `MoneyAgent`.

EwoutH commented:
Personally I won't bother with updating the old spaces, we will replace them soon anyways.

DarshPareek commented:
> Personally I won't bother with updating the old spaces, we will replace them soon anyways.
> 
> @DarshPareek Could you make your PR title and description a bit more clear/descriptive?

Thank you for the suggestion! I'll revise the PR title and description to better reflect the changes made. Please let me know if there’s anything specific you'd like included.

EwoutH commented on .pre-commit-config.yaml:
```diff
@@ -32,3 +32,7 @@ repos:
         "--ignore-words",
         ".codespellignore",
       ]
+      exclude: >
```
Sorry, I totally missed this in earlier reviews. Why are we excluding all notebooks from Codespell?

If there’s a certain word you would like to use, you can add it to [`.codespellignore`](https://github.com/projectmesa/mesa/blob/main/.codespellignore)

quaquel replied:
Could you please remove all changes to .codespellignore?

DarshPareek replied:
If i don't add words to .codespellignore the pre-commit check fails on the branch, If I remove them is there any workaround the check?

EwoutH replied:
This still needs to be cleaned up

DarshPareek replied:
I have cleaned it up.

EwoutH replied:
Final things done in f4b7eb6.

EwoutH commented on .codespellignore:
```diff
@@ -8,3 +8,10 @@ fpr
+COO
+Onl
+Te
+ND
```
Sorry, this is a bit nitpicky, but could you explain why all these “words” need to be added?

DarshPareek replied:
Okay so the codespell is flagging these words as errors they are present in the notebook, i think they are the part of image data that is used. 
![Screenshot 2024-12-14 at 19-29-46 ](https://github.com/user-attachments/assets/b8e1628b-df6f-4a9d-afda-abab679a2e7b)
here "COO" is considered as a word so it it flagged by codespell and thus the branch fails the pre-commit test.
If there is another workaround this I will gladly update the branch

EwoutH replied:
Okay, that clears up a lot.

I would propose just cleaning the cell outputs.

DarshPareek commented:
I have cleared the outputs and removed the words from .codespellignore, let me know if any further changes are needed.

EwoutH commented:
Cleanup the final things, merging.

Thanks @DarshPareek!

DarshPareek commented:
> Cleanup the final things, merging.
> 
> Thanks @DarshPareek!

Thanks! I would love to contribute in future.
